### PR TITLE
shared: add virtual entity types

### PR DESCRIPTION
**Pull request by deluvas1911**

### CODE REVIEW

File: client/index.d.ts [hunk 1]
```diff
@@ -391,6 +397,55 @@ declare module "alt-client" {
...
+    public getSyncedMeta<V extends any>(key: string): V | undefined;
+
+    /**
+     * Determines whether contains the specified key.
+     *
+     * @param key The key of the value to locate.
+     * @returns True if the meta table contains any data at the specified key or False if not
+     */
+    public hasSyncedMeta(key: string): boolean;
+    public hasSyncedMeta<K extends shared.ExtractStringKeys<shared.ICustomEntitySyncedMeta>>(key: K): boolean;
```
Comment (xxshady) [inline client/index.d.ts:444]: `ICustomEntitySyncedMeta` is used by `alt.Entity`, for virtual entity you should declare another interface

And is it synced meta? I think is should be stream synced meta:
https://github.com/altmp/cpp-sdk/blob/80c4d58e813daf764dbdeb371e26f0019f5d0609/script-objects/IVirtualEntity.h#L31-L33

so the name of the interface should be: `ICustomVirtualEntityStreamSyncedMeta`

File: server/index.d.ts [hunk 2]
```diff
@@ -620,6 +620,82 @@ declare module "alt-server" {
...
+     *
+     * @remarks The given value will be shared with all clients in streaming range.
+     *
+     * @param key The key of the value to store.
+     * @param value The value to store.
+     */
+    public setStreamSyncedMeta<K extends string>(key: K, value: shared.InterfaceValueByKey<shared.ICustomEntityStreamSyncedMeta, K>): void;
+    public setStreamSyncedMeta<K extends shared.ExtractStringKeys<shared.ICustomEntityStreamSyncedMeta>>(key: K, value: shared.ICustomEntityStreamSyncedMeta[K]): void;
+    /** @deprecated See {@link "alt-shared".ICustomEntityStreamSyncedMeta} */
+    public setStreamSyncedMeta<V extends any, K extends string = string>(key: K, value: shared.InterfaceValueByKey<shared.ICustomEntityStreamSyncedMeta, K, V>): void;
```
Comment (xxshady) [inline server/index.d.ts:688]: https://github.com/altmp/altv-types/pull/236/files#r1148633286

File: server/index.d.ts [hunk 3]
```diff
@@ -620,6 +620,82 @@ declare module "alt-server" {
...
+
+    /**
+     * Removes the specified key and the data connected to that specific key.
+     *
+     * @param key The key of the value to remove.
+     */
+    public deleteStreamSyncedMeta(key: string): void;
+    public deleteStreamSyncedMeta<K extends shared.ExtractStringKeys<shared.ICustomEntityStreamSyncedMeta>>(key: K): void;
```
Comment (xxshady) [inline server/index.d.ts:696]: https://github.com/altmp/altv-types/pull/236/files#r1148633286

**Comment by xxshady** [comment]
stream synced meta methods are fine now

File: client/index.d.ts [hunk 4]
```diff
@@ -391,6 +397,55 @@ declare module "alt-client" {
...
+    public static readonly all: ReadonlyArray<VirtualEntity>;
+
+    /** Unique id */
+    public readonly id: number;
+
+    /** Virtual Entity Group */
+    public readonly group: VirtualEntityGroup;
+
+    /** Unique clientside id */
+    public readonly remoteId: number;
```
Comment (xxshady) [inline client/index.d.ts:424]: `Unique clientside id` why it's clientside if it's "remote" id

File: server/index.d.ts [hunk 5]
```diff
@@ -620,6 +620,82 @@ declare module "alt-server" {
...
+    /** Returns all VirtualEntity instances */
+    public static readonly all: ReadonlyArray<VirtualEntity>;
+
+    /** Unique id */
+    public readonly id: number;
+
+    /** Virtual Entity Group */
+    public readonly group: VirtualEntityGroup;
+
+    /** Unique clientside id */
```
Comment (C0kkie) [inline server/index.d.ts:652]: i am not sure if that is a clientside id lmao

**Pull request closed without merging**